plugins/modules/k8s_scale.py

```diff
@@ -271,22 +271,25 @@ def _continue_or_exit(warn):
             continue
 
         if existing_count != replicas:
-            if not module.check_mode:
-                if module.params["kind"] == "job":
-                    existing.spec.parallelism = replicas
-                    result = resource.patch(existing.to_dict()).to_dict()
+            if module.params["kind"] == "job":
+                existing.spec.parallelism = replicas
+                result = {"changed": True}
+                if module.check_mode:
+                    result["result"] = existing.to_dict()
                 else:
-                    result = scale(
-                        module,
-                        k8s_ansible_mixin,
-                        resource,
-                        existing,
-                        replicas,
-                        wait,
-                        wait_time,
-                        wait_sleep,
-                    )
-                    changed = changed or result["changed"]
+                    result["result"] = resource.patch(existing.to_dict()).to_dict()
+            else:
+                result = scale(
+                    module,
+                    k8s_ansible_mixin,
+                    resource,
+                    existing,
+                    replicas,
+                    wait,
+                    wait_time,
+                    wait_sleep,
+                )
+                changed = changed or result["changed"]
```

Comment:
I guess you can do something like this if you want` changed |= result["changed"]`

Reply:
yes we can do that but this is not like a requirement depending on the reviwer people may like this one
So I keep it like that for now